### ISSUE TITLE
"Query optimizer" like design?

### ISSUE THREAD

**Issue by gergness** [gergness/srvyr#8]
Since: 
```
svymean(~x+y, svy)
```

is faster than:
```
svymean(~x, svy)
svymean(~y, svy)
```

Maybe `summarize()` should capture all variables that are having the same operation done, and then do them together.

**Comment by carlganz** [comment]
Interesting. I didn't realize that the `survey` functions scaled so well. I did a little benchmarking and with a 4.4MB `svrepdesign` object. `svymean`s with a single input take around 18 to 35 milliseconds, while `svymean` with 7 inputs takes around 130 milliseconds (18 milliseconds per variable), so it seems stabler. Using `Rcpp`, I was able to get the time [under 4 milliseconds](https://github.com/carlganz/surveyRcpp/blob/master/vignettes/Intro.pdf), but that was just for replicate weights, so implementing the query optimizer seems worthwhile. Regards
